src/condor_utils/dagman_utils.cpp

```diff
@@ -768,14 +802,13 @@ DagmanUtils::RenameRescueDagsAfter(const std::string& primaryDagFile, bool multi
 void
 DagmanUtils::tolerant_unlink(const std::string &pathname)
 {
+	if ( ! fileExists(pathname)) { return; }
```

Comment:
This does not prevent ENOENT, sadly; some other random process on the same machine could call `unlink()` between this check and the next.  You could revert back to the old code, except with `errnor != ENOENT` no special-case warning code?

Reply:
I figured this doesn't fully prevent `ENOENT`. I would rather treat a race condition deletion as an error message than a warning message at this point. It is a tolerant unlink and we detected the file existed prior to attempting to unlink so we should inform an error (In my opinion)

Reply:
Also, I added the check so for file not existing at the top to prevent getting lots of failed to remove file because file does not exist messages for `condor_submit_dag` with a fresh DAG file and directory.